src/entities/contract-agreement.ts

```diff
@@ -1,12 +1,19 @@
 import { Policy } from "./policy";
+import { JsonLdId, JsonLdValue } from "./jsonld"
 
-export interface ContractAgreement {
-  id: string;
-  assetId: string;
-  consumerAgentId: string;
+export class ContractAgreement extends JsonLdId {
+  'https://w3id.org/edc/v0.0.1/ns/assetId': JsonLdValue<string>[];
+  consumerAgentId?: string;
   contractEndDate?: number;
   constractSigningDate?: number;
   constractStartDate?: number;
-  policy: Policy;
-  providerAgentId: string;
+  policy?: Policy;
+  providerAgentId?: string;
+
+  get assetId(): string {
+    return this['https://w3id.org/edc/v0.0.1/ns/assetId']
```

Comment:
@ndr-brt do you think it will make sense if we tried to use `${EDC_CONTEXT}assetId` instead of always hardcoding the value? it seems like there is a pattern in getting those values in general, it's usually 'https://w3id.org/edc/v0.0.1/ns/entityName'

Reply:
hey, I'm working at this, let's check the latest commit :volcano:

Reply:
The idea is to keep the objects not typed and defining properties through `get` methods that take info out of the json-ld properties

Reply:
mmmh... despite this is a really clean approach, typescript lacks this feature: https://github.com/microsoft/TypeScript/issues/14417
and it's pretty fundamental for what I was doing here... maybe something that can be used in the meantime for "output" objects, instead for creating "input" objects better go with the current approach.
The intersection between the two is the `Policy` class, that's also a crucial one.